File: БР1.1/Захматов Юрий/labs/lab2/property-service/src/main/java/com/petproject/propertyservice/entities/PropertyImageEntity.java
package com.petproject.propertyservice.entities;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;


@Table(name = "property_image")
@NoArgsConstructor
@Getter
@Setter
@Entity
@Builder
@AllArgsConstructor
public class PropertyImageEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "property_id" ,nullable = false)
    private PropertyEntity propertyId;

    private String url;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = this.createdAt;
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: БР1.2/Михайлов_Юрий/labs/lab3/property_service/src/models/property-image.entity.ts
import { BaseEntity, Column, Entity, PrimaryGeneratedColumn } from 'typeorm';

@Entity()
export class PropertyImage extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ type: 'int', nullable: false })
    property_id: number;

    @Column({ type: 'varchar', length: 1000, nullable: false })
    url: string;
}

